File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-0.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        first = 0
        flag = False
        second = 1
        while second < len(nums):
            if nums[first]!=nums[second]:
                if flag:
                    first+=1
                    nums[first] = nums[second]
                else:
                    first+=1
                
                
            else:
                flag = True

            second+=1
        return first + 1

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> freq;
        vector<pair<int,int>> freqlist;
        vector<int> result;

        for (int x: nums)
        {
            freq[x]++;
        }
        for(auto &p: freq)
        {
            freqlist.push_back({p.first, p.second});
        
        }
        sort(freqlist.begin(), freqlist.end(),[](auto &a, auto &b){
            return a.second> b.second;
        });

        for(int i=0;i<k;i++)
        {
            result.push_back(freqlist[i].first);
        }
       
    return result;
    }
    
};
